subgraphs/compound-forks/src/mapping.ts

```diff
@@ -1590,25 +1589,25 @@ export function getOrElse<T>(
 //
 // create seperate InterestRate Entities for each market snapshot
 // this is needed to prevent snapshot rates from being pointers to the current rate
-function getSnapshotRates(rates: string[], identifier: string): string[] {
+function getSnapshotRates(rates: string[], timeSuffix: string): string[] {
   let snapshotRates: string[] = [];
   for (let i = 0; i < rates.length; i++) {
-    let actualRate = InterestRate.load(rates[i]);
-
-    // get/create new snapshot rate
-    let rateId = actualRate!.side + "-" + actualRate!.type + "-" + identifier;
-    let _rate = InterestRate.load(rateId);
-    if (!_rate) {
-      _rate = new InterestRate(rateId);
-      _rate.rate = BIGDECIMAL_ZERO;
-      _rate.side = actualRate!.side;
-      _rate.type = actualRate!.type;
+    let rate = InterestRate.load(rates[i]);
+    if (!rate) {
+      log.warning("[getSnapshotRates] rate {} not found, should not happen", [
+        rates[i],
+      ]);
+      continue;
     }
 
-    // update rate to current rate
-    _rate.rate = actualRate!.rate;
-    _rate.save();
-    snapshotRates.push(_rate.id);
+    // create new snapshot rate
+    let snapshotRateId = rates[i].concat(timeSuffix);
```

Comment:
this will work, but we probably want a hyphen between the rate id and `timeSuffix`